app/services/fetch_coronavirus_statistics_service.rb

```diff
@@ -68,21 +80,135 @@ def parse_response(response)
-      parsed[:cumulative_vaccinations] = latest_vaccinations["cumulativeVaccinations"]
-      parsed[:cumulative_vaccinations_date] = Date.parse(latest_vaccinations["date"])
+    parsed.merge!(latest_vaccinations(data))
+    parsed.merge!(latest_admissions(data))
+    parsed.merge!(latest_tests(data))
+
+    parsed
+  end
+
+  def latest_vaccinations(data)
+    latest_vaccinations = data.find do |d|
+      d["cumulativeFirstDoseVaccinations"] &&
+        d["cumulativeSecondDoseVaccinations"] &&
+        d["percentageFirstVaccine"] &&
+        d["percentageSecondVaccine"]
     end
 
-    if (latest_admissions = data.find { |d| d["hospitalAdmissions"] })
-      parsed[:hospital_admissions] = latest_admissions["hospitalAdmissions"]
-      parsed[:hospital_admissions_date] = Date.parse(latest_admissions["date"])
+    return {} unless latest_vaccinations
+
+    {
+      cumulative_first_dose_vaccinations: latest_vaccinations["cumulativeFirstDoseVaccinations"],
+      cumulative_second_dose_vaccinations: latest_vaccinations["cumulativeSecondDoseVaccinations"],
+      percentage_first_vaccine: latest_vaccinations["percentageFirstVaccine"],
+      percentage_second_vaccine: latest_vaccinations["percentageSecondVaccine"],
+      cumulative_vaccinations_date: Date.parse(latest_vaccinations["date"]),
+    }
+  end
+
+  def latest_admissions(data)
+    latest_admissions = data.find { |d| d["hospitalAdmissions"] }
+    return {} unless latest_admissions
+
+    daily_admissions = daily_admissions(latest_admissions)
+    weekly_admissions = weekly_admissions(latest_admissions, data)
+
+    daily_admissions.merge!(weekly_admissions)
+  end
+
+  def daily_admissions(latest_admissions)
+    {
+      hospital_admissions: latest_admissions["hospitalAdmissions"],
+      hospital_admissions_date: Date.parse(latest_admissions["date"]),
+    }
+  end
+
+  def weekly_admissions(latest_admissions, data)
+    weeks_with_admissions = []
+    data.each do |day_admissions|
+      weeks_with_admissions << day_admissions["hospitalAdmissions"] if on_or_before?(day_admissions["date"], latest_admissions["date"])
     end
 
-    if (latest_tests = data.find { |d| d["newPositiveTests"] })
-      parsed[:new_positive_tests] = latest_tests["newPositiveTests"]
-      parsed[:new_positive_tests_date] = Date.parse(latest_tests["date"])
+    total_current_week_admissions = current_week_admissions(weeks_with_admissions)
+    total_previous_week_admissions = previous_week_admissions(weeks_with_admissions)
+
+    return {} unless total_current_week_admissions && total_previous_week_admissions
+
+    {
+      current_week_hospital_admissions: total_current_week_admissions,
+      current_week_hospital_admissions_change_number: total_current_week_admissions - total_previous_week_admissions,
+      current_week_hospital_admissions_change_percentage: percentage_change(total_current_week_admissions, total_previous_week_admissions),
```

Comment:
How are these percentage changes calculated? Here's the current numbers on my local:

![Screenshot 2021-11-03 at 10 13 02](https://user-images.githubusercontent.com/861310/140042875-dd5a9e8d-52a8-4e92-ae29-75011f22ae32.png)

Doing the maths myself, 10.4% of 280,479 is `29169.816`, and 2.5% of 7,197 is `179.925`, which doesn't match the numbers above, although my maths isn't great and maybe this is being calculated in a different way?

Reply:
```
((previous weeks total - current week total) / previous weeks total) * 100
```

Reply:
Thou shalt find that the percentages on coronavirus.data.gov.uk matcheth mine 😎

Reply:
Oh I see. So it's calculating it based on the difference between this weeks total (shown) and last weeks total (not shown). That makes sense, thanks.